Link UI: polish lightbox pieces.

jasmussen opened the pull request:
## What?

This addresses as yet unfinished design work that went into #54916. Specifically the followup items mentioned here: https://github.com/WordPress/gutenberg/pull/57608#pullrequestreview-1835879214

The metrics in the link UI for lightbox can be improved:

![lighbox ui before](https://github.com/WordPress/gutenberg/assets/1204802/fec4b33b-b024-4f94-84d4-333c4c511935)

and the focus style is broken

![focus style broken](https://github.com/WordPress/gutenberg/assets/1204802/b5b84146-be4f-470a-887d-91743d8920aa)

This PR fixes both. First it addresses the metrics:

![new metrics](https://github.com/WordPress/gutenberg/assets/1204802/3a36e2b8-e432-456c-b025-df1d3b0b292a)

* 48px tall control, same as the block toolbar
* Harmonious padding and gap
* Tweaked vertical alignment
* Correct helper text font size

And the focus style is repaired too:

![fixed focus style](https://github.com/WordPress/gutenberg/assets/1204802/6b054afa-48a4-408c-9012-64a20c4f46ca)

## Testing Instructions

Insert an image, test the URL focus, test the lighbox option.

jasmussen committed:
Link UI: polish lightbox pieces.

t-hamano commented:
There is #58505 which makes a similar adjustment. #58505 also tries to add a default border to the input field and adjust the layout when opening link settings.

Is it possible to mix these two types of PR and try a more ideal approach?

cc @afercia

jasmussen commented:
I'd be happy to pull out any conflicting changes and make this only about the URL pieces.

richtabor commented:
> There is https://github.com/WordPress/gutenberg/pull/58505 which makes a similar adjustment. https://github.com/WordPress/gutenberg/pull/58505 also tries to add a default border to the input field and adjust the layout when opening link settings.

> Is it possible to mix these two types of PR and try a more ideal approach?

I like that this effort is simpler and actionable; a quick fix, not a refactor. 

As I detailed https://github.com/WordPress/gutenberg/pull/58505#issuecomment-1927153440, we should follow-up with using LinkControl everywhere, as detailed in https://github.com/WordPress/gutenberg/issues/50893.

richtabor reviewed (approved):
LGTM

t-hamano committed:
Polish info style

t-hamano commented:
I see, so let's merge this PR first and follow up with #58505.

I made a slight adjustment with [cbbc4f4](https://github.com/WordPress/gutenberg/pull/58666/commits/cbbc4f4cd5178fdf69bc2a01b400d6639a89b663). I removed the icon wrapper div and removed unnecessary class names and styles. For the description, I followed [the style of the menu item](https://github.com/WordPress/gutenberg/blob/cbbc4f4cd5178fdf69bc2a01b400d6639a89b663/packages/components/src/menu-item/style.scss#L61-L63), added a 5px margin-top and changed the text color from `$gray-600` to `$gray-700`.

Before:

![before](https://github.com/WordPress/gutenberg/assets/54422211/8f885745-d31c-4fa6-ae55-d8c9ecfdef07)

After:

![after](https://github.com/WordPress/gutenberg/assets/54422211/c4006826-3eb0-4fd9-9492-e64c1cf11f5f)

Let me know what you think of this.

jasmussen commented:
Thanks for all the reviews, and thanks for pushing. Please feel free to both continue pushing, and to press the green button when you're ready. It isn't clear to me what is the best order to merge these changes, so I'm happy to defer to you.

> I made a slight adjustment with [cbbc4f4](https://github.com/WordPress/gutenberg/pull/58666/commits/cbbc4f4cd5178fdf69bc2a01b400d6639a89b663). I removed the icon wrapper div and removed unnecessary class names and styles. For the description, I followed [the style of the menu item](https://github.com/WordPress/gutenberg/blob/cbbc4f4cd5178fdf69bc2a01b400d6639a89b663/packages/components/src/menu-item/style.scss#L61-L63), added a 5px margin-top and changed the text color from $gray-600 to $gray-700.

While I do appreciate the code cleanup (feel free to merge, we can always polish later), the line-height was intentional. Specifically, we need the title (Expand on click) and the helper text (Scales the image...) to not be taller than 32px at most. When that's the case, the entire box is a total of 48px tall which matches exactly the height of the block toolbar. This gives some shared DNA in terms of the "materials" used. 

As noted, not a blocker, and it might be nice to look at title+helptext in a more holistic way. From what I can tell, even after these changes the metrics are not the same as they are in this dropdown:

![Screenshot 2024-02-06 at 11 41 56](https://github.com/WordPress/gutenberg/assets/1204802/8a70be6f-b063-42a2-a340-e6332e6262f8)

t-hamano commented:
Thank you for the detailed explanation. Perhaps the focus is on this margin-top?

https://github.com/WordPress/gutenberg/pull/58666/files#diff-49bee8a5008288296ba2faca8ec5cb424a7c7d56790d40cf7f28cc6966409662R83

If you don't mind, is it possible to revert the unintended changes I made? Of course, you are welcome to revert my entire commit 👍

jasmussen commented:
I have to step away for some dinner prep and such, if you are able to do the revert, feel free to, otherwise I'll look tomorrow. But also as noted, it seems fine to land as is, I'm happy to follow up on the height separately if need be.

t-hamano committed:
Remove top margin from description

t-hamano reviewed (approved):
> While I do appreciate the code cleanup (feel free to merge, we can always polish later), the line-height was intentional. Specifically, we need the title (Expand on click) and the helper text (Scales the image...) to not be taller than 32px at most. When that's the case, the entire box is a total of 48px tall which matches exactly the height of the block toolbar. This gives some shared DNA in terms of the "materials" used.

Thanks for letting me know why! I removed the top margin of the helper text. I think it matches the original state of this PR, with the only exception being a change in the color of the helper text.

I think we can ship this PR, but if I've missed anything please comment.

jasmussen commented:
I'll merge, and I'll follow up, just let me know.

afercia commented:
Noting that this PR doesn't have an associated issue. It's not the only case though. It appears that PRs with no accompanying issue are a recurring trend in this project. I would like to remind everyone that [the guidelines of the Gutenberg project](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/repository-management.md#pull-requests) and the spirit that always inspired contribution to the WordPress.org project require any _non-trivial pull requests to be preceded by a related issue_. Allowing for broader discussion, in an issue or a Trac ticket, is fundamental in an open-source, collaborative, project.

I'm not sure ignoring that guideline encourages contribution to this project in any way and I'm not sure it's in line with the principles of the WordPress.org project. It also triggers the perception that there is a 'double standard' in the Gutenberg project, where some contributors feel they can do whatever they want while other contributors are asked to follow a more strict process. Lastly, I'm not sure this is the most efficient way to contribute, as it leads to overlapping issues and PRs.

That said, I'm not sure a closed PR is the best place for this kind of conversation. I'll consider to move these considerations to a more appropriate place.

jasmussen commented:
Oh you're right. I've updated the issue to include the pertinent issue and conversation.

afercia commented:
> Oh you're right. I've updated the description to include the pertinent issue and conversation.

If I'm not wrong, the issue you linked to: https://github.com/WordPress/gutenberg/issues/54916 was closed as completed on Jan 31, 2024. This PR was created on Feb 5, 2024 so it's actually a PR associated to a _closed_ issue. For the future, in such cases I'd kindly ask everyone to either reopen the supposedly associated issue or create a new one, please.

Also, neither this PR nor #54916 nor [the PR]([url](https://github.com/WordPress/gutenberg/pull/57608)) with the mentioned [code review comment](https://github.com/WordPress/gutenberg/pull/57608#pullrequestreview-1835879214) have any accessibility related GitHub label, while they do impact important accessibility features like the inputs border and focus style.

jeryj commented on packages/block-editor/src/components/url-input/style.scss:
```diff
@@ -15,11 +15,13 @@ $input-size: 300px;
 		margin-right: 0;
 
+		&:not(:focus) {
+			border-color: transparent;
```
This caused the text link editing popover to have a transparent border on the link combobox:

<img width="355" alt="Link Control with no border on input" src="https://github.com/WordPress/gutenberg/assets/967608/64ed0ee1-a2a6-4fae-9ba9-22bb924e1ba2">

t-hamano replied:
Thank you for finding that. Perhaps the increase in CSS selector specificity for adding transparent borders caused this border to disappear unintentionally. In #58505, restoring the default border of the URL input field, i.e. removing this style itself, should resolve this issue.

![image](https://github.com/WordPress/gutenberg/assets/54422211/1146382f-75a0-4f10-95e9-d7412593ec9b)
